### DISCUSSION TITLE
Incorrect nuget dependencies are identified

### DISCUSSION THREAD

**Discussion by iamrahul127** [DependencyTrack/dependency-track#1755] [General]
Analysis of project referring NServiceBus.Newtonsoft.Json v 2.3.0 results in to identifying incorrect dependency. We expect DT to identify Newtonsoft.Json v 13.0.1 but it's identifying 11.0.2 version of instead as given in below screenshot. This results in finding incorrect vulnerability. 

```
.NETFramework 4.5.2
  [Newtonsoft.Json](>= 13.0.1 && < 14.0.0)
  [NServiceBus] (>= 7.0.0 && < 8.0.0)

.NETStandard 2.0
  [Newtonsoft.Json] (>= 13.0.1 && < 14.0.0)
  [NServiceBus] (>= 7.0.0 && < 8.0.0) 
```
### Current Behavior:
![image](https://user-images.githubusercontent.com/24877322/175931889-d2a2b8b8-c956-4f71-91ea-7df54325a491.png)

### Expected Behavior:
Should identify following as dependencies instead 
![image](https://user-images.githubusercontent.com/24877322/175932220-29951400-c63d-4682-b031-6ab90e61f1b6.png)

### Environment:
  
  - Dependency-Track Version:  v4.4.1
  - Distribution: Docker 
  - BOM Format & Version: CycloneDX 1.3
  - Database Server: PostgreSQL
  - Browser: Chrome

**Comment by nscuro** [comment]
Hi @iamrahul127 👋

Dependency-Track works with data it's being fed with the BOM you upload. This looks like the tool you used to generate the BOM with produced a BOM with different contents than you expected.

What generator were you using to produce the BOM? Is it [cyclonedx-dotnet](https://github.com/CycloneDX/cyclonedx-dotnet) per chance? If so, you may want to file a defect over there.

**Reply by iamrahul127** [reply]
Hi @nscuro, 
Yes, we are using [cyclonedx-dotnet](https://github.com/CycloneDX/cyclonedx-dotnet). I have investigated further and look like it's a bug in [cyclonedx-dotnet](https://github.com/CycloneDX/cyclonedx-dotnet). 
Sample solution with two projects look like below and attached to the thread.

![image](https://user-images.githubusercontent.com/24877322/176167520-b8ae8ddf-349f-46ca-b3b9-fd651747b00b.png)

**Observations:**

1. When I generate BOM for entire solution using `dotnet CycloneDX -o . -j BOMIssue.sln` following dependency is added to BOM. 
` {
      "ref": "pkg:nuget/NServiceBus.Newtonsoft.Json@2.3.0",
      "dependsOn": [
        "pkg:nuget/NServiceBus@7.3.0",
        "pkg:nuget/Newtonsoft.Json@11.0.2"
      ]
    },`

2. When I generate BOM only for  using `dotnet CycloneDX -o . -j BOMIssue/BOMIssue.csproj` following dependency is added to BOM.

` {
      "ref": "pkg:nuget/**NServiceBus.Newtonsoft.Json@2.3.0**",
      "dependsOn": [
        "pkg:nuget/NServiceBus@7.0.0",
        "pkg:nuget/**Newtonsoft.Json@13.0.1**"
      ]
    },`

I think tool is reporting incorrect dependency of **NServiceBus.Newtonsoft.Json@2.3.0** on **Newtonsoft.Json@11.0.2**. In fact **IdentityModel@4.1.1** is depending on  **Newtonsoft.Json@11.0.2**.

Could you please investigate further?
[BOMIssue.zip](https://github.com/DependencyTrack/dependency-track/files/9000549/BOMIssue.zip)

**Reply by nscuro** [reply]
Please raise this issue in the cyclonedx-dotnet repository (https://github.com/CycloneDX/cyclonedx-dotnet), so its maintainers can investigate.
